GameOfRunes/GameOfRunes/GameEngine/GameMetaData.swift

```diff
@@ -20,14 +20,26 @@ class GameMetaData {
     var playerMana: Int
     var score: Int
     var selectedPowerUp: PowerUpType?
+    var levelWaves: EnemySpawnUnit
+    var levelSpawnInterval: Double
+    var numEnemiesOnField: Int
 
-    init(maxPlayerHealth: Int, numManaUnits: Int, manaPerManaUnit: Int, powerUps: [PowerUpType]) {
+    init(maxPlayerHealth: Int, numManaUnits: Int, manaPerManaUnit: Int,
+         powerUps: [PowerUpType], levelNumber: Int) {
         self.maxPlayerHealth = maxPlayerHealth
         self.numManaUnits = numManaUnits
         self.manaPerManaUnit = manaPerManaUnit
         availablePowerUps = powerUps
         playerHealth = maxPlayerHealth
         playerMana = 0
         score = 0
+        numEnemiesOnField = 0
+
+        do {
+            levelWaves = try LevelCreator.getLevelData(levelNumber: levelNumber)
+            levelSpawnInterval = try LevelCreator.getLevelSpawnInterval(levelNumber: levelNumber)
```

Comment:
Is this repeated computation?

Reply:
wait but aren't static variables just basically singletons. There's only a single copy of it. So there won't be recreation of the level, i believe.

Reply:
@andyylam seems to be right. @brianyenna, the static var are computed variables. So it will compute every time it is called. Use `static var level1 = { <computation closure> } ()` and it will make it lazy and also be computed only once.

Reply:
This is more like a static function rather than static variable (correct me if I'm wrong, nor exactly sure the exact difference between computed vars and functions)

Reply:
computed var is a cleaner way to write functions that take in no params, doesn't mutate state and returns a value. Basically a getter that requires no params.

Reply:
TIL hahaha. Okay will refactor. See latest commit